build/prepend_edoc_autogenerated

```diff
@@ -1,27 +1,28 @@
-#! /bin/sh
+#!/usr/bin/env escript
+%% -*- erlang -*-
 
-f="$1"
-orig="${2:-$f}"
-
-if (head -n 1 "$f" | grep -q '@doc automatically generated scanner')
-then
-    # already prepended
-    exit 0
-fi
-
-cleanup () {
-    rm -f "$f".tmp
-}
-
-trap 'ec=$?; cleanup; exit $ec' EXIT QUIT INT TERM
-
-set -e
-cat > "$f".tmp <<EOF
-%% @doc automatically generated scanner
+-define(HEAD_LINE, "@doc automatically generated scanner").
+-define(ALL_AUTO_STR_FORMAT, "%%" ++ ?HEAD_LINE ++ "
 %% @private
 %% @end
--file("$orig", 0).
-EOF
-cat "$f" >> "$f".tmp
-touch -r "$f" "$f".tmp
```

Comment:
Need to set timestamp on the `ErlFile` to avoid unncessary rebuilding, use `file:read_file_info` and `file:write_file_info`.